### ISSUE TITLE
Allowed memory exhausted when installing v5

### ISSUE THREAD

**Issue by Furo42** [ffraenz/private-composer-installer#34]
When I try to install the package "private-composer-installer" v5 I get the following error message:

```
- Installing ffraenz/private-composer-installer (v5.0.0): Extracting archive
PHP Fatal error:  Allowed memory size of 8589934592 bytes exhausted (tried to allocate 4294967304 bytes) in %PATH_TO_PROJECT%\vendor\ffraenz\private-composer-installer\src\PrivateComposerInstaller\Environment\LoaderFactory.php on line 47

Fatal error: Allowed memory size of 8589934592 bytes exhausted (tried to allocate 4294967304 bytes) in %PATH_TO_PROJECT%\vendor\ffraenz\private-composer-installer\src\PrivateComposerInstaller\Environment\LoaderFactory.php on line 47

Check https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for more info on how to handle out of memory errors.
```

I already tried to increase the memory limit up to 8 GB but the error still occurs.

My Environment:
Windows 10 Pro 2004
PHP 7.2.28 is installed via XAMPP

This problems happens with composer 1.10.16 and composer 2.0.3
Installing v4 works without problems.

**Comment by ffraenz** [comment]
Hi @Furo42, I found an issue that may lead to an endless loop when dealing with Windows paths. Are you able to test the patch in your specific environment to make sure it fixes the issue?

For that you would need to require the package version `dev-dev`. To allow development versions you need to add the following lines to your `composer.json`. Make sure you undo these changes after testing.

```
"minimum-stability": "dev",
"prefer-stable": true
```

Let me know if this works for you. I'll prepare a patch release if so.

**Comment by Furo42** [comment]
Hi @ffraenz ,
I tested the fix both with composer 1 and 2 and it worked fine without any problems.

Thank you for the fast response and I'm looking forward to the new release!

**Comment by ffraenz** [comment]
@Furo42 Thank you for your quick feedback! I just released [v5.0.1](https://github.com/ffraenz/private-composer-installer/releases/tag/v5.0.1). You should now be good to go.
